### PR TITLE
sst_importer: don't cache rewritten files

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/15483

What's Changed:
This PR make cache in memory caches keys not rewritten.

```commit-message
The rewrite step of sst_importer::apply has been delayed to while iterating the file.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Fixed a bug that may cause data loss while sequently execute multi `restore point` within one minute.
```

### CODE REVIEW

File: components/tikv_util/src/codec/stream_event.rs [hunk 1]
```diff
@@ -42,14 +62,45 @@ impl EventIterator<'_> {
...
+        self.offset += key_len;
+    }
+
+    fn move_to_next_key_with_rewrite(&mut self) {
+        let key_len = self.get_size() as usize;
+        let rewrite = self.rewrite_rule.as_ref().expect("rewrite rule not set");
+        if key_len < rewrite.from.len()
+            || &self.buf[self.offset..self.offset + rewrite.from.len()] != rewrite.from
+        {
+            self.move_to_next_key_without_rewrite();
```
Comment (Leavrth) [inline components/tikv_util/src/codec/stream_event.rs:81]: if runs to here, there must be some wrong. how about print the log?

File: components/tikv_util/src/codec/stream_event.rs [hunk 2]
```diff
@@ -42,14 +62,45 @@ impl EventIterator<'_> {
...
+            return;
+        }
+        self.key_buf.clear();
+        self.key_buf.extend_from_slice(rewrite.to);
```
Comment (Leavrth) [inline components/tikv_util/src/codec/stream_event.rs:85]:
```suggestion
        self.key_buf.reserve(rewrite.to.len() + key_len - rewrite.from.len());
        self.key_buf.extend_from_slice(rewrite.to);
```

**Review by Leavrth — APPROVED** [review]

File: components/tikv_util/src/codec/stream_event.rs (hunk 1 above)
Comment (YuJuncen) [reply components/tikv_util/src/codec/stream_event.rs:81]: I think we can check this outside🤔

**Review by 3pointer — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**